### ISSUE TITLE
Does not support PHP constants as textdomains

### ISSUE THREAD

**Issue by geminorum** [felixzapata/gulp-checktextdomain#3]
often I'm using `PLUGIN_TEXTDOMAIN` instead of `'plugin-textdomain'`, that would be nice.

p.s. [wp-pot](https://github.com/rasmusbe/wp-pot) support this.

**Comment by felixzapata** [comment]
hi, what do you need exactly? In the wp-pot plugin I can´t find nothing related with the `PLUGIN_TEXTDOMAIN

**Comment by geminorum** [comment]
instead of using: 
```
_ex( 'String', 'Context', 'my-plugin-textdomain' ); 
```
we use php constants: 
```
define( 'MY_PLUGIN_TEXTDOMAIN', 'my-plugin-textdomain' );
_ex( 'String', 'Context', MY_PLUGIN_TEXTDOMAIN );
```
and in `wp-pot` options:
```
{
    package: "My Plugin",
    domain: "MY_PLUGIN_TEXTDOMAIN"
}
```

**Comment by felixzapata** [comment]
can you make a pull request with the update you need?

**Comment by felixzapata** [comment]
I 've checked to the `wp-pot`. If I understand ok, you can pass the domain you need via options:

```
wpPot({
  destFile: 'file.pot',
  **domain: 'domain',**
  package: 'Example project',
  src: 'src/*.php'
});
```
Any string you pass in the options will be used as domain. The same is for this plugin, you pass the domain via string with the options:

```
text_domain: 'my-domain', //Specify allowed domain(s)
```

**Comment by felixzapata** [comment]
i´ve found the test where `wp-pot` handle strings with domain set as a constant. I will check if I can do the same. 

Anyway, if you can make the pull request with the update, I will appreaciate

**Comment by geminorum** [comment]
I think the main difference that is `wp-pot` uses [glayzzle/php-parser](https://github.com/glayzzle/php-parser) and checks the [constant ref](https://github.com/glayzzle/php-parser/blob/master/docs/AST.md#constref).

It seems awfully a lot to ask to switch to `php-parser`.

thanks anyway.

**Comment by felixzapata** [comment]
I saw yesterday that there is the [same issue in the grunt plugin](https://github.com/stephenharris/grunt-checktextdomain/issues/6). 

Sincerely, I have no idea how difficult is to switch to the other parser.

**Comment by ichiriac** [comment]
Hi @felixzapata, 

You can start by using the `php-parser` tokenizer by replacing the checktextdomain.js with something like :

```js
'use strict';
var parser = require('php-parser');
module.exports = {
  token_get_all: function(source) {
    return parser.tokenGetAll(source);
  }
};
```

This parser produces exactly the same output as the PHP `token_get_all` function. but as I can see [here](https://github.com/felixzapata/gulp-checktextdomain/blob/master/index.js#L149) you try to analyze the source in order to extract informations, best way to do this, is to work directly on the AST (code source is transformed to a structured object representation), something like :

```js
var extractions = [];
function visit(node) {
  // generic extractor
  if (node.type === 'call' && node.what.name === '__') {
    extractions.push(node);
  }
  // recursive scan
  if (Array.isArray(node.children)) node.children.forEach(visit);
  if (Array.isArray(node.body)) node.body.forEach(visit);
}

// here the bootstrap
var parser = require('php-parser');
visit(parser.parseCode(sourceCode));
```

You can take a look and take a look at the AST structure here : https://astexplorer.net/#/gist/c5098fd3b19f3c0da9de29986aa553d9/578cf3e85d8b5e4aee923e3fa3a24e9de93def19

**Comment by felixzapata** [comment]
hi @ichiriac , thanks. Are you able to make a pull request with the needed changes?

**Comment by ichiriac** [comment]
ok @felixzapata, it should be easy to bootstrap with `php-parser`, just a few lines of code, but I will need you to test and maybe adapt some code as I don't have enough experience in building modules on gulp and using gettext.

**Comment by felixzapata** [comment]
no worry about that. I will test the pull request and the code

**Comment by ichiriac** [comment]
it's done. I did not tested yet with constants, but in principle it should work. It was cool you had already a test suite in order to check changes :+1:

**Comment by felixzapata** [comment]
Thanks. I will take a look and I will add the new test I had for this new feature.

**Comment by felixzapata** [comment]
I've released version 2.0.0. Thanks @ichiriac and @geminorum.
